blockpage.lua

```diff
@@ -0,0 +1,156 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Ring Zero Desenvolvimento de Software LTDA
+-- SPDX-License-Identifier: GPL-2.0-only
+--
+
+local linux  = require("linux")
+local socket = require("socket")
+local pack   = require("dome/pack")
+local config = require("dome/config")
+
+local PACKET = socket.af.PACKET
+local RAW    = socket.sock.RAW
+
+local ETH_HLEN  = 14
+local ETH_P_ALL = 0x0003
+
+local IPPROTO_TCP = 0x06
+
+local RST = 1 << 2
+local ACK = 1 << 4
+
+local packer   = pack.packer
+local unpacker = pack.unpacker
+
+local blockpage = {}
+
+-- https://datatracker.ietf.org/doc/html/rfc1141
+local function checksum(data, size)
+	local sum = 0
+	for i = 1, size, 2 do
+		sum = sum + string.unpack(">I2", data, i)
+	end
+	if size % 2 == 1 then
+		sum = sum + string.unpack("I1", data, size)
+	end
+	sum = sum & 0xffffffff
+	while sum >> 16 ~= 0 do
+		sum = (sum >> 16) + sum & 0xffff
+	end
+	return ~sum & 0xffff
+end
+
+local function ethhdr(packet)
+	local str, be16 = unpacker(packet)
```

Comment:
just for completeness, as we are calling `set*` when we use `packer`, I would call `get*` when we use `unpacker`